### ISSUE TITLE
Error while updating property 'accessibilityState' of a view managed by : RCTView , null ,   Value of disabled cannot be cast from ReadableNativeMap to Boolean

### ISSUE THREAD

**Issue by mrshahzeb7** [facebook/react-native#35119]
### New Version

0.70.4

### Old Version

0.68.2

### Build Target(s)

platform :ios, '12.4'

### Output of `react-native info`

>>>>>>>>>npx react-native info<<<<<<<<<<

info Fetching system and libraries information...
System:
    OS: macOS 11.4
    CPU: (4) x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
    Memory: 104.48 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.2 - ~/node_modules/.bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.17.0 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 28, 29, 30, 31, 33
      Build Tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0
      System Images: android-28 | Google APIs Intel x86 Atom_64, android-29 | Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-32 | Google APIs Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8139111
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.14.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0
    react-native: 0.70.4 => 0.70.4
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Issue and Reproduction Steps

we faced this while upgrading react native version from 0.68.2 to 0.70.4
we also tested by creating a new project with the latest version it also shows the same error

This happens when we pass an object to disabled property in TouchableOpacity

 <TouchableOpacity
      onPress={onPress}
      disabled={ {} }                           //it will show error
      disabled={ {} ? false : true }    //it won't show error
 </TouchableOpacity>
    
Now the problem is that we are using this type of condition on the whole app  (disabled={item?.address}) and we have to change it everywhere

**Comment by mrshahzeb7** [comment]
<img width="326" alt="MicrosoftTeams-image (9)" src="https://user-images.githubusercontent.com/69900372/198580634-12cab8a8-0448-4ecb-b279-447d8f3efbbb.png">

**Comment by aryella-lacerda** [comment]
Something very similar started happening to me after I updated to 0.70:

<img src="https://user-images.githubusercontent.com/28730324/203771098-c2ebb739-5e11-4125-89d3-e73a976fffde.png" width="350"/>

The problem was that my button looked like this:

 ```tsx
     <Button
        label={'Next'}
        disabled={!phone || !country || error} // <--- error was an empty string
      />
```

I switched it to:

 ```tsx
     <Button
        label={'Next'}
        disabled={!phone || !country || !!error} // <--- "!!error" is now a boolean
      />
```

**Comment by viraj-glide** [comment]
So, Button, TouchableOpacity, Pressable which uses disabled.It takes a Boolean or Falsey / Truthy values. We can convert the value using !! operator

**Comment by mehdiChibani** [comment]
> Something very similar started happening to me after I updated to 0.70:
> 
> <img alt="" width="350" src="https://user-images.githubusercontent.com/28730324/203771098-c2ebb739-5e11-4125-89d3-e73a976fffde.png">
> 
> The problem was that my button looked like this:
> 
> ```tsx
>     <Button
>        label={'Next'}
>        disabled={!phone || !country || error} // <--- error was an empty string
>      />
> ```
> 
> I switched it to:
> 
> ```tsx
>     <Button
>        label={'Next'}
>        disabled={!phone || !country || !!error} // <--- "!!error" is now a boolean
>      />
> ```

that was the same issue for me , solved by setting disabled field with a boolean value

**Comment by milan-vala** [comment]
i was having the same issue, was not explicitly checking if value is `null`.
`<TouchableOpacity
    disabled={registeredDevices[`module${index + 1}`]}
    onPress={() => handleModuleClick(index + 1)}
>
    <Text>{index + 1}</Text>
</TouchableOpacity>`

so i made it
`<TouchableOpacity
    disabled={registeredDevices[`module${index + 1}`] !== null}
    onPress={() => handleModuleClick(index + 1)}
>
    <Text>{index + 1}</Text>
</TouchableOpacity>`

**Comment by HamidAliSE** [comment]
Yes! TouchableOpacity's disabled prop accept a bool value. If you need to calculate this bool value with a function, you should invoke the function in the disabled prop like the example below.

```
<TouchableOpacity **_disabled={isDisabled()}_**
<Text >Hamid Ali</Text>
</TouchableOpacity>
```

I hope, it might help someone.

**Comment by ArsaCore** [comment]
In my case, I was using Nativewind and the issue was "transition-all duration-300" in className.
